lib/github.atd

```diff
@@ -264,9 +288,9 @@ type api_commit = {
   sha: commit_hash;
   commit: inner_commit;
   html_url <ocaml name="url"> : string;
-  author: github_user;
-  files: file list;
-  stats: api_commit_stats;
+  ?author: github_user option;
```

Comment:
I'm not entirely sure why but on the return of the compare URL, the outside `commit_api` payload have author as null but the inner commit payload within the `commit_api` payload has:
![image](https://user-images.githubusercontent.com/23148113/199377104-3cd4ecc6-7b4b-43ad-a7d9-bc6064afddec.png)

Reply:
You can see they have different types. The outer one is a GitHub user (`github_user`), while inner one is the git commit's author (`git_user_with_date`). See:
https://docs.github.com/en/rest/commits/commits

It looks like in your case you get outer author = null because the email you used to commit the file isn't registered with any GH account. Please add a comment to the field explaining that.

(I'm surprised this hasn't caused issues before)

Reply:
ah amazing thanks so much @yasunariw!